### PR TITLE
oiiotool -i extensions that make channel extraction efficient

### PR DESCRIPTION
**Pull request by lgritz**
This is spawned by the observation in issue #1537 that extracting a small number of channels from a many-channeled file is very expensive. This is exacerbated when the image is large, has a huge number of channels, not all channels having the same data type, and the file is untiled.

In #1537, the situation was a 50 channel scanline EXR that was a mixture of half and float, resolution 5760x2880, and a simple extraction of the R,G,B,A channels took 1:34. 

This set of patches accomplishes the following:
- Improvements to the logic that handles per-channel format conversion when reading scanline files: (a) use threads; (b) convert batches of adjacent channels that have the same data type, instead of handling each channel individually.
- ImageBuf::read() is extended with a new variety that lets you populate the IB with a (contiguous) subset of channels existing in the file being read. This always saves the buffer memory of the unneeded channels and the computation required to do any data format conversions of those channels, and for some image file formats might also save disk I/O (it does for OpenEXR, but not for most file formats).
- oiiotool option `-i:type=...` explicitly overrides the internal datatype, rather than converting internally to float (which is good for speeding up math computation on pixels, but not helpful when just shuffling channels around).
- oiiotool option `-i:ch=...` reads only the listed channels of the input. I works just like having a normal input and then a separate `--ch` command, except that by knowing at input time which channels you want, it has the potential to save all the buffer memory, type conversion, and data copying of the unneeded channels, and for some formats (EXR - yes! but not most others) can also save I/O for the unneeded channels.

| time | peak mem | command |
| --- | --- | --- |
| 1:34 | 3.4 GB | **original**: `oiiotool input.exr -ch R,G,B,A -d half -o out.exr` |
|  |  | _after per-channel logic improvement:_ |
| 0:32 | same | `oiiotool input.exr -ch R,G,B,A -d half -o out.exr` |
| 0:10 | 0.2 GB | `oiiotool -i:type=half input.exr -ch R,G,B,A -d half -o out.exr` |
| 0:02 | 0.4 GB | `oiiotool -i:ch=R,G,B,A input.exr -d half -o out.exr` |
| 0:01 | 0.2 GB | `oiiotool -i:type=half:ch=R,G,B,A input.exr -d half -o out.exr` |

Bottom line: on my machine, this improves time by at least 60x and memory by 15x. Pretty good for one (very very late) evening's work.

Let that be a lesson to you: giving me a specific example file + command line that performs slower than you'd like can sometimes result in really big improvements!

### CODE REVIEW

**Comment by lgritz** [comment]
@mike1158 Ping?

**Comment by lgritz** [comment]
@mike1158 ?

**Comment by mike1158** [comment]
Sorry for taking so long to get back to you.  I applied the patch and get effectively the same results you do.  Thanks again!
-miker
